openmc/data/multipole.py

```diff
@@ -129,6 +143,600 @@ def _broaden_wmp_polynomials(E, dopp, n):
+            #TODO: improve the metric with clearer basis
+            quality = ratio + ratio2 - min(0.1*maxre, 1) - 0.001*new_poles.size
+
+            if np.any(test_xs < -atol):
+                quality = -np.inf
+
+            if log >= DETAILED_LOGGING:
+                print("  # poles: {}".format(new_poles.size))
+                print("  Max relative error: {:.3f}%".format(maxre*100))
+                print("  Satisfaction: {:.1f}%, {:.1f}%".format(ratio*100, ratio2*100))
+                print("  Quality: {:.2f}".format(quality))
+
+            if quality > best_quality:
+                if log >= DETAILED_LOGGING:
+                    print("  Best by far!")
+                found_better = True
+                best_quality, best_ratio = quality, ratio
+                best_poles, best_residues = new_poles, residues
+                best_test_xs, best_relerr = test_xs, relerr
+                if best_ratio >= 1.0:
+                    if log:
+                        print("Found ideal results. Stop!")
+                    found_ideal = True
+                    break
+            else:
+                if log >= DETAILED_LOGGING:
+                    print("  Discarded!")
+
+        if found_ideal:
+            break
+
+        # acceleration
+        if found_better:
+            n_discarded = 0
+        else:
+            if order > max(2*n_peaks, 50) and best_ratio > 0.7:
+                n_discarded += 1
+                if n_discarded >= 10 or (n_discarded >= 5 and best_ratio > 0.9):
+                    if log >= DETAILED_LOGGING:
+                        print("Couldn't get better results. Stop!")
+                    break
+
+    # merge conjugate poles
+    real_idx = []
+    conj_idx = []
+    found_conj = False
+    for i, p in enumerate(best_poles):
+        if found_conj:
+            found_conj = False
+            continue
+        if np.imag(p) == 0.:
+            real_idx.append(i)
+        else:
+            if i < best_poles.size and np.conj(p) == best_poles[i + 1]:
+                found_conj = True
+                conj_idx.append(i)
+            else:
+                raise RuntimeError("Complex poles are not conjugate!")
+    if log:
+        print("Found {} real poles and {} conjugate complex pairs.".format(
+               len(real_idx), len(conj_idx)))
+    mp_poles = best_poles[real_idx + conj_idx]
+    mp_residues = np.concatenate((best_residues[:, real_idx],
+                                  best_residues[:, conj_idx]*2), axis=1)/1j
+    if log:
+        print("Final number of poles: {}".format(mp_poles.size))
+
+    if path_out:
+        if not os.path.exists(path_out):
+            os.makedirs(path_out)
+        for i, mt in enumerate(mts):
+            fig, ax1 = plt.subplots()
+            lns1 = ax1.loglog(test_energy, test_xs_ref[i], 'g', label="ACE xs")
+            lns2 = ax1.loglog(test_energy, best_test_xs[i], 'b', label="VF xs")
+            ax2 = ax1.twinx()
```

Comment:
For a bunch of nuclides, I get the following at this line:
```
openmc/data/multipole.py:366: UserWarning: Data has no positive values, and therefore cannot be log-scaled.
```

Reply:
Can you give me an example nuclide which showed the warning so I can repeat? I probably encountered the same warning before but forgot the nuclides.

Reply:
Looks like H3 with search turned off should do it. Here's a reproducer:
```Python
openmc.data.WindowedMultipole.from_endf(
    '/opt/data/endf/endf-b-viii.0/neutrons/n-001_H_003.endf',
    log=True,
    vf_options={'path_out': '.'},
    wmp_options={'search': False}
)
```

Reply:
Yeah, checks added. I also changed `with np.errstate(invalid='ignore'):` to `with np.errstate(invalid='ignore', divide='ignore'):` so the running on `H3` should be clean now.